src-java/base-topology/base-storm-topology/src/test/java/org/openkilda/wfm/topology/TestKafkaConsumer.java

```diff
@@ -92,6 +104,45 @@ public String pollMessageValue() throws InterruptedException {
         return Optional.ofNullable(pollMessage()).map(ConsumerRecord::value).orElse(null);
     }
 
+    /**
+     * Polls messages from kafka until expectedCount is reached.
+     *
+     * @param expectedCount expected count of messages
+     * @param clazz         class of messages
+     * @return list of messages
+     */
+    public <T> List<T> assertNAndPoll(final int expectedCount, Class<T> clazz) {
+        List<T> messages = new ArrayList<>();
+        ConsumerRecord<String, String> record = null;
+        for (int i = 0; i < expectedCount; i++) {
+            try {
+                record = pollMessage(kafkaMessagePollTimeout);
+                if (record == null) {
+                    throw new AssertionError(format("Could not get %d records. %d records gotten",
+                            expectedCount, messages.size()));
+                }
+                messages.add(MAPPER.readValue(record.value(), clazz));
+            } catch (InterruptedException e) {
+                throw new AssertionError(format("Could not get %d records. %d records gotten", expectedCount,
+                        messages.size()));
+            } catch (IOException e) {
+                throw new AssertionError(format("Could not parse %s object: '%s'", clazz.getName(), record.value()));
+            }
+        }
+        try {
+            // ensure that we received exact expected count of records
+            record = pollMessage(100);
```

Comment:
100 is timeout? Assume there is more messages than expected. If it fast enough it throws an exception, if it is slow it just returns the messages. This doesn't seem correct. (or I am missing something?)

Reply:
Yes, this is to check that there are no remaining messages. I don't understand why it isn't correct